PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/parameters.spec.ts

```diff
@@ -0,0 +1,349 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { instance, mock } from 'ts-mockito'
+
+import ConsoleWrapper from '../../wrappers/consoleWrapper'
+import Parameters from '../../updaters/parameters'
+import TaskLibWrapper from '../../wrappers/taskLibWrapper'
+import async from 'async'
+import { expect } from 'chai'
+
+describe('parameters.ts', (): void => {
+  let taskLibWrapper: TaskLibWrapper
+  let consoleWrapper: ConsoleWrapper
+
+  beforeEach((): void => {
+    taskLibWrapper = mock(TaskLibWrapper)
+    consoleWrapper = mock(ConsoleWrapper)
+  })
+  describe('initialize', (): void => {
+    describe('initializer function', (): void => {
+      it('initialize - should give all default values', (): void => {
+        // Arrange
+        const parameters: Parameters = new Parameters(instance(consoleWrapper), instance(taskLibWrapper))
+
+        // // Act
+        parameters.initialize('', '', '', '', '') // NaN
+
+        // // Assert
+        expect(parameters.baseSize).to.equal(250)
+        expect(parameters.growthRate).to.equal(2.0)
+        expect(parameters.testFactor).to.equal(1.5)
+        expect(parameters.fileMatchingPatterns).to.deep.equal(['**/*'])
+        expect(parameters.codeFileExtensions.length).to.equal(108)
```

Comment:
It would be great to check the debug and console log statements are printed as well